docs/rules/require-meta-docs-recommended.md

```diff
@@ -0,0 +1,39 @@
+# Require rules to implement a `meta.docs.recommended` property (`eslint-plugin/require-meta-docs-recommended`)
+
+<!-- end auto-generated rule header -->
+
+Defining a whether recommended value for each rule can help developers understand whether they're recommended.
```

Comment:
I would expand on the pros and cons here to help developers understand if the rule makes sense for them:

```suggestion
Utilizing `meta.docs.recommended` makes it clear from each rule implementation whether a rule is part of the `recommended` config. Some plugins also have scripting for conveniently generating their config based on this flag.

However, this flag may not be appropriate for all plugins:

* Extra scripting/tooling is needed to keep the flags in sync with the config
* The flag may not scale to plugins that have multiple/many configs or don't have a recommended config
* Or some may simply prefer to keep the source of truth solely in the config rather than duplicating config membership data in rules
```